app/controllers/passwords_controller.rb

```diff
@@ -54,4 +80,33 @@ def deliver_email(user)
     mail = ::ClearanceMailer.change_password(user)
     mail.deliver_later
   end
+
+  def setup_mfa_authentication
+    return if @user.mfa_disabled?
+    @form_mfa_url = mfa_edit_user_password_url(@user, token: @user.confirmation_token)
+  end
+
+  def setup_webauthn_authentication
+    return if @user.webauthn_credentials.none?
+
+    @form_webauthn_url = webauthn_edit_user_password_url(@user, token: @user.confirmation_token)
+
+    @webauthn_options = @user.webauthn_options_for_get
+
+    session[:webauthn_authentication] = {
+      "challenge" => @webauthn_options.challenge
+    }
+  end
+
+  def mfa_edit_conditions_met?
+    @user.mfa_enabled? && @user.otp_verified?(params[:otp])
+  end
+
+  def login_failure(message)
+    @form_url = mfa_edit_user_password_url(@user, token: @user.confirmation_token)
```

Comment:
```suggestion
    @form_mfa_url = mfa_edit_user_password_url(@user, token: @user.confirmation_token)
```
Is there a way to also check that it was correctly set on the page in the functional test (eg. `page.has_content(<mfa_edit_url>)`?